### ISSUE TITLE
URL with double slashes within an onclick element highlighted as a code comment

### ISSUE THREAD

**Issue by CoconutMacaroon** [microsoft/vscode#202915]
Does this issue occur when all extensions are disabled?: Yes/No
Yes. I did `code-insiders.cmd --disable-extensions`, and the issue still happened then.

- VS Code Version: `1.86.0-insider`
- OS Version: Windows 11 Pro 22H2

Steps to Reproduce:

1. Create a new HTML file, such as `demo.html`
2. Paste in the following HTML code:
   ```html
   <!DOCTYPE html>
   <html lang="en">
   <head>
       <title>Hello world!</title>
       <script>
           console.log("https://example.com");
       </script>
   </head>
   <body>
       <button onclick="console.log('https://example.com');">Click me!</button>
   </body>
   </html>
   ```
3. Observe the rendering of the text:
   ![image](https://github.com/microsoft/vscode/assets/45187468/6bf4cc9f-3625-4c8c-b8ba-9e9f9e487faa)

The string renders correctly when in a `<script> ... </script>` tag. However, when placed within an `onclick` of a `<button> ... </button>`, it (incorrectly) highlights the two slashes and the remainder of the string in green, as if it was a code comment. Interestingly, the semicolon is also green.

The expected behavior is that VS Code recognizes that it isn't a comment, but rather part of a string, and so highlights it like it does in the `<script> ... </script>` tag.

**Comment by perplexyves** [comment]
Hello @CoconutMacaroon. A workaround to this colorization issue would be to treat the slash **/** as if it was an escape character:
```html
<button onclick="console.log('https:\/\/example.com');">Click me!</button>
```
It would be nice though to dig deeper in it and find out why it's behaving like this.
Thank you for noticing!

**Comment by perplexyves** [comment]
Hello @aeschli. Thanks for pointing out to the library textmate/html.tmbundle. However, the PR fix for this issue has been opened there 3 years ago and still hasn't been merged yet. Can we do something regarding it?

**Comment by aghArdeshir** [comment]
I agree with @perplexyves . The latest commit/issue/MR in that repository is 3 years old. According to the WikiPedia of TextMate:

> In October 2021, the latest/last commit was made to GitHub, though no announcement has been made to the effect that the project has been officially abandoned and its source code is still publicly available.

https://en.wikipedia.org/wiki/TextMate

Looks like dead to me too :+1:

**Comment by alexr00** [comment]
Reopened and repurposed this issue to track finding a new grammar for HTML syntax highlighting. My first look didn't turn up any good options.

**Comment by aghArdeshir** [comment]
@alexr00 I was preparing an issue about this last night, I left it half finished as draft, but posted it tonight: https://github.com/microsoft/vscode/issues/203212

Now I noticed you've repurposed this issue for that. Feel free to close what I've just created, but I've left some more information about the problem there.
